pkg/controller/storagecluster/storagecluster.go

```diff
@@ -774,6 +774,12 @@ func (c *Controller) manage(
 		}
 	}
 
+	if cluster.Status.Phase == "" && cluster.Spec.CloudStorage.MaxStorageNodesPerZone == nil {
```

Comment:
why not let it change? new zones may be added later

Reply:
Once the value is set, we don't have insight into if the customer modified it or it's the default value. May be the customer want's to continue to retain the value that exists. Since we are not in a position to determine that we will not let it change. 
If it's an existing cluster, we don't want to meddle with what the customer already has.
The only scenario we want to handle for an existing cluster is upgrade. I'll work on that and will send out a PR in a week.
The point of the default value is not to add too much intelligence instead just cover some obvious cases where we can start introducing the option with a non-zero value.

Reply:
Like Piyush said, this wont be persisted